### PR TITLE
feat: training files

### PR DESCRIPTION
**Pull request by vanessavirgo94**
## Description

Description of what this PR does. What have you added or changed, and why?  If it fixes a bug or resolves a feature request, be sure to link to that issue.

This is a new code repository that will help new users of the platform to become enablement. It links to tutorials here https://platformsupport.peak.ai/a/solutions/articles/80001038072?portalId=80000082893

## Review Checks

Please check if the PR fulfills these requirements:

_Put an `x` in the boxes that apply, Remove any lines that do not apply_
- [x] 📝 The commit message is clear and descriptive
- [x] 🔐 The Security Considerations section in the PR description is complete - **Please do not remove this**
- [x] ✅ Tests for the changes have been added and run successfully including the new changes
- [x] 📄 Documentation has been added / updated (for bug fixes / features)

## Dependencies

Add links to any pull requests or documentation related to this pull request.

None

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
- [ ] Yes

## Security Considerations

Are there any other security or data concerns to be aware of?

Please discuss the security implications/considerations relevant to the proposed change. 
This may include...
* security-relevant design decisions
* concerns 
* important discussions
* implementation-specific guidance and pitfalls
* an outline of threats and risks and how they are being addressed.

- No credentials hard coding
- Not injecting vars in SQL queries 

## Types of change

What kind of change does this Pull Request introduce?
_Put an `x` in the boxes that apply_

- [ ] 🐛 Bugfix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🛠 Adding or updating configuration files, development scripts etc.
- [ ] ♻️ Refactoring (no functional changes, no api changes)
- [ ] 🧹 Chore (removing redunant files, fixing typos etc.)
- [x] 📄 Documentation Update 
- [x] ❓ Other (if none of the other choices apply)

### CODE REVIEW

File: api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+COPY ./app.py /app/app.py
+COPY ./run.sh /app/run.sh
+WORKDIR /app
+
+RUN useradd demo
+USER demo
+
+EXPOSE 8080
+
+ENTRYPOINT ["bash", "run.sh"]
```
Comment (azhar22k) [inline api/Dockerfile]:
```suggestion
ENTRYPOINT ["bash", "run.sh"]

```

File: api/app.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Import your packages 
+from flask import Flask, Response, jsonify, request
+import pickle
+import boto3
+import pandas as pd
+import os
+from snowflake.connector.pandas_tools import write_pandas
+import snowflake.connector
```
Comment (azhar22k) [inline api/app.py]:
```suggestion
import pickle
import boto3
import os
import snowflake.connector

import pandas as pd

from snowflake.connector.pandas_tools import write_pandas
from flask import Flask, Response, jsonify, request
```

It is a good practice to group similar imports

File: api/app.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    return Response(response, status=200)
+
+# Health route to check the API is running
+@app.route("/health")
+def health():
+    return Response("OK", status=200)
+
+
+if __name__ == "__main__":
+    app.run(host="0.0.0.0", port=8080)
```
Comment (azhar22k) [inline api/app.py]:
```suggestion
    app.run(host="0.0.0.0", port=8080)
   
```

File: api/run.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash 
+
+gunicorn app:app --bind 0.0.0.0:8080 --workers=4
```
Comment (azhar22k) [inline api/run.sh]:
```suggestion
gunicorn app:app --bind 0.0.0.0:8080 --workers=4

```

File: workflows/r/untitled.txt [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+# Building and fitting Linear Regression model. You can change this
+print("Build model")
+linreg_reg_spec <- 
+  linear_reg() %>% 
+  set_engine("lm")
+
+linreg_reg_fit <- linreg_reg_spec %>% fit(price ~ ., data = data)
+
+# Predicting on our data and creating a column called predictions
```
Comment (peak-jiteshahuja) [inline workflows/r/untitled.txt]: Hey @vanessavirgo94 - Do we want to keep the file name as untitled ? Seems a bit odd to me.

File: workflows/python/workflowscript.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
+# First import the packages that will be needed  
+from snowflake.connector.pandas_tools import write_pandas
+import snowflake.connector
+import pandas as pd
+from sklearn.linear_model import LinearRegression
+import boto3
+import pickle
+import os
+import tempfile
```
Comment (azhar22k) [inline workflows/python/workflowscript.py]:
```suggestion
import boto3
import pickle
import os
import tempfile
import snowflake.connector

import pandas as pd

from snowflake.connector.pandas_tools import write_pandas
from sklearn.linear_model import LinearRegression
```

File: workflows/python/workflowscript.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+# Saving our model to our datalake to use later
+print("Save model to s3")
+s3 = boto3.client("s3")
+with tempfile.TemporaryFile() as fp:
+    pickle.dump(model, fp)
+    fp.seek(0)
+    s3.upload_fileobj(fp, 
+                      Bucket=os.environ['DATA_LAKE'],
+                      Key=key_model_save)
```
Comment (azhar22k) [inline workflows/python/workflowscript.py]:
```suggestion
                      Key=key_model_save)
                      
```

File: api/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,49 @@
+# Python version 
+FROM python:3.7
```
Comment (peak-jiteshahuja) [inline api/Dockerfile:2]: Hey @vanessavirgo94 - Just checking if the version should be 3.8 as thats what we been using at Peak ?

File: workspace/Pulling data from a database (R).R [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+# Load the packages you need
+library(DBI)
+library(odbc)
+library(tidyverse)
+
+# Create a connection string using the `odbc` file within a Workspace
+conn <- dbConnect(odbc(), paste0(Sys.getenv('TENANT') , '-' , Sys.getenv('STAGE')))
+
+# Pull the data from Snowflake into a Workspace
+df <- dbGetQuery(conn, “select * from STAGE.HOUSEPRICES”)
```
Comment (azhar22k) [inline workspace/Pulling data from a database (R).R]:
```suggestion
df <- dbGetQuery(conn, “select * from STAGE.HOUSEPRICES”)

```

File: api/Dockerfile (hunk 1 above)
Comment (peak-jiteshahuja) [inline api/Dockerfile]: Hey @vanessavirgo94 - Just thinking if this should be part of https://github.com/peak-ai/platform-global-images ?

File: workflows/r/untitled.txt (hunk 5 above)
Comment (vanessavirgo94) [reply workflows/r/untitled.txt]: No we dont! Sorry I will change

File: api/Dockerfile (hunk 1 above)
Comment (vanessavirgo94) [reply api/Dockerfile]: Nope, its an API and it wouldn't make sense

Comment (vanessavirgo94) [reply api/Dockerfile]: Everyones API is different

File: api/app.py [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
```
Comment (peak-jiteshahuja) [inline api/app.py]: Nit. Newline.

File: workflows/python/workflowscript.py [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+    cur = conn.cursor()
+    cur.execute(query)
+    df = cur.fetch_pandas_all()
+    return df
+
+# Printing statements so we have some logs
+print("Querying Snowflake data")
+
+# Query to get all of our raw data from STAGE
+query = """
```
Comment (peak-jiteshahuja) [inline workflows/python/workflowscript.py:42]:
```suggestion
query = "select * from STAGE.HOUSEPRICES"
```

File: api/app.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
```
Comment (azhar22k) [inline api/app.py:47]:
```suggestion
   
   
```

File: workflows/python/workflowscript.py [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+# Saving our model to our datalake to use later
+print("Save model to s3")
+s3 = boto3.client("s3")
+with tempfile.TemporaryFile() as fp:
+    pickle.dump(model, fp)
+    fp.seek(0)
+    s3.upload_fileobj(fp, 
+                      Bucket=os.environ['DATA_LAKE'],
+                      Key=key_model_save)
+
```
Comment (azhar22k) [inline workflows/python/workflowscript.py:143]:
```suggestion
                      
                      
```

File: workflows/python/workflowscript.py [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+# Printing statements so we have some logs
+print("Querying Snowflake data")
+
+# Query to get all of our raw data from STAGE
+query = """
+select * from STAGE.HOUSEPRICES
+"""
+conn = connect_to_db()
+df = read_from_db(conn, query)
```
Comment (peak-jiteshahuja) [inline workflows/python/workflowscript.py:46]: Nit. Variable name `df` could have been better.

File: api/Dockerfile (hunk 8 above)
Comment (vanessavirgo94) [reply api/Dockerfile:2]: Leave this for now

File: workflows/python/workflowscript.py (hunk 14 above)
Comment (vanessavirgo94) [reply workflows/python/workflowscript.py:46]: Leave this please

File: workspace/Pulling data from a database (R).R [hunk 15]
```diff
@@ -0,0 +1,15 @@
+## Install the connector you want to use 
+install.packages(‘DBI’)
```
Comment (peak-jiteshahuja) [inline workspace/Pulling data from a database (R).R:2]: Hey @vanessavirgo94 - Shouldn't we be placing these files under `python` and `R` folders as we have done for `workflows` ?

**Review by azhar22k — APPROVED** [review]

**Review by peak-jiteshahuja — APPROVED** [review]

**Pull request merged**